spec/sidekiq/evss/disability_compensation_form/submit_form0781_spec.rb

```diff
@@ -81,6 +84,257 @@
+                               saved_claim_id: saved_claim.id,
+                               form_json: form0781,
+                               submitted_claim_id: evss_claim_id)
+    end
+    let(:perform_upload) do
+      subject.perform_async(submission.id)
+      described_class.drain
+    end
+    let(:path_to_0781_fixture) { 'spec/fixtures/pdf_fill/21-0781/simple_0781.pdf' }
+    let(:sample_0781_file_read) { File.read(path_to_0781_fixture) }
+    let(:path_to_0781a_fixture) { 'spec/fixtures/pdf_fill/21-0781a/simple.pdf' }
+    let(:sample_0781a_file_read) { File.read(path_to_0781a_fixture) }
+    let(:parsed_0781_form) { JSON.parse(submission.form_to_json(Form526Submission::FORM_0781))['form0781'] }
+    let(:parsed_0781a_form) { JSON.parse(submission.form_to_json(Form526Submission::FORM_0781))['form0781a'] }
+
+    before do
+      Flipper.enable(:disability_compensation_use_api_provider_for_0781_upload)
+      # StatsD metrics are incremented in several callbacks we're not testing here so we need to allow them
+      allow(StatsD).to receive(:increment)
+    end
+
+    context 'when the disability_compensation_upload_0781_to_lighthouse flipper is enabled' do
+      let(:faraday_response) { instance_double(Faraday::Response) }
+      let(:lighthouse_request_id) { Faker::Number.number(digits: 8) }
+      let(:expected_statsd_metrics_prefix) do
+        'worker.evss.submit_form0781.lighthouse_supplemental_document_upload_provider'
+      end
+      let(:lighthouse_0781_document) do
+        LighthouseDocument.new(
+          claim_id: submission.submitted_claim_id,
+          participant_id: submission.auth_headers['va_eauth_pid'],
+          document_type: 'L228'
+        )
+      end
+      let(:lighthouse_0781a_document) do
+        LighthouseDocument.new(
+          claim_id: submission.submitted_claim_id,
+          participant_id: submission.auth_headers['va_eauth_pid'],
+          document_type: 'L229'
+        )
+      end
+
+      before do
+        Flipper.enable(:disability_compensation_upload_0781_to_lighthouse)
+
+        allow(BenefitsDocuments::Form526::UploadSupplementalDocumentService).to receive(:call)
+          .and_return(faraday_response)
+
+        allow(faraday_response).to receive(:body).and_return(
+          {
+            'data' => {
+              'success' => true,
+              'requestId' => lighthouse_request_id
+            }
+          }
+        )
+      end
+
+      it 'uploads the 0781 documents to Lighthouse' do
+        # 0781
+        allow_any_instance_of(described_class).to receive(:generate_stamp_pdf).with(parsed_0781_form,
+                                                                                    submission.submitted_claim_id, '21-0781').and_return(path_to_0781_fixture)
+
+        allow_any_instance_of(LighthouseSupplementalDocumentUploadProvider).to receive(:generate_upload_document).and_return(lighthouse_0781_document)
+
+        allow(File).to receive(:delete).and_return(nil)
```

Comment:
This is another one. Yeah I feel like it's such an anti pattern to have the test know all of these internal details/we have to maintain this, but I think it's just a problem with the design of all of the document uploads